### PR TITLE
titanium-data-table && titanium-data-table-item

### PR DESCRIPTION
**Pull request by aarondrabeck**

### CODE REVIEW

File: src/titanium-data-table.ts [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+        this.selected.length !== this.items.length}">
+        <svg checked viewBox="0 0 24 24">
+          <path d="M0 0h24v24H0z" fill="none" />
+          <path d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z" />
+        </svg>
+      </div>
+      <div title="Clear selection" ?hidden="${
+        this.selected.length === 0 ||
+        this.selected.length === this.items.length}">
+        <svg indetermanite viewBox="0 0 24 24">
```
Comment (antwonthegreat) [inline src/titanium-data-table.ts]: Does indetermanite do anything? It is spelled wrong

File: src/titanium-data-table.ts [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+  updated(changedProps) {
+    if (changedProps.has('items') && changedProps.get('items') !== this.items) {
+      // Clear selection when items array changes.
+      this.clearSelection();
+    }
+  }
+
+  private _handleItemSelectionChange(
+      e: CustomEvent<{item: any, isSelected: boolean}>) {
+    if (e.detail.isSelected) {
```
Comment (antwonthegreat) [inline src/titanium-data-table.ts]: Since this event has bubbles, should we stop it here?

File: src/titanium-data-table.ts [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+      this.requestUpdate();
+      this._notifiySelectedChanged();
+    } else {
+      this.selected.splice(this.selected.indexOf(e.detail.item), 1);
+      this.requestUpdate();
+      this._notifiySelectedChanged();
+    }
+  }
+
+  private _notifiySelectedChanged() {
```
Comment (antwonthegreat) [inline src/titanium-data-table.ts]: spelling notifiy

File: demo/titanium-data-table-demo.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+    demo-container {
+        @apply --layout-vertical;
+        padding: 16px;
+    }
+</style>
+<demo-container>
+
+    <h4>Table Options</h4>
+    <paper-toggle-button checked="{{singleSelect}}">Multi Select</paper-toggle-button>
```
Comment (antwonthegreat) [inline demo/titanium-data-table-demo.ts]: should be {{!singleSelect}}

File: demo/titanium-data-table-demo.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+            sort-direction="{{sortDirection}}"></titanium-table-header>
+        <titanium-table-header slot="table-headers" column-name="last" title="Last Name" sort-by="{{sortBy}}"
+            sort-direction="{{sortDirection}}"></titanium-table-header>
+        <titanium-table-header slot="table-headers" desktop column-name="title" title="Title" sort-by="{{sortBy}}"
+            sort-direction="{{sortDirection}}"></titanium-table-header>
+        <titanium-table-header slot="table-headers" desktop center column-name="amount" title="Centered" sort-by="{{sortBy}}"
+            sort-direction="{{sortDirection}}"></titanium-table-header>
+        <titanium-table-header slot="table-headers" desktop right column-name="amount" title="Right" sort-by="{{sortBy}}"
+            sort-direction="{{sortDirection}}"></titanium-table-header>
+        <titanium-table-header slot="table-headers" width="40px" column-name="amount" title="Width" sort-by="{{sortBy}}"
```
Comment (antwonthegreat) [inline demo/titanium-data-table-demo.ts]: When the header is this small the sort iron is squished down and looks weird
titaniumTable
d

File: src/titanium-data-table.ts [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+
+@customElement('titanium-data-table')
+export class TitaniumDataTable extends LitElement {
+  @property() title: string;
+  @property() take: number;
+  @property() page: number = 0;
+  @property() count: number;
+  @property() items: Array<any>;
+  @property({type: Boolean, attribute: 'single-select', reflect: true})
+  singleSelect: boolean;
```
Comment (antwonthegreat) [inline src/titanium-data-table.ts]: If singleSelect is changed from the parent and multiple items are selected, deselectAll should probably be called

File: src/titanium-data-table.ts (hunk 1 above)
Comment (aarondrabeck) [reply src/titanium-data-table.ts]: It doesn't, removed.

File: src/titanium-data-table.ts (hunk 2 above)
Comment (aarondrabeck) [reply src/titanium-data-table.ts]: Yep good idea.

File: src/titanium-data-table.ts (hunk 6 above)
Comment (aarondrabeck) [reply src/titanium-data-table.ts]: Not worried about this, single select isn't made to be toggled during use.  I dont see a use case.  Let me know if you can think of a use case.

File: demo/titanium-data-table-demo.ts (hunk 5 above)
Comment (aarondrabeck) [reply demo/titanium-data-table-demo.ts]: Issue with demo.  Fixed.

**Comment by aarondrabeck** [comment]
@antwonthegreat  Comments addresses, thx

**Pull request merged**